backend/utils.py

```
@@ -183,29 +183,82 @@ def extract_diy_metadata(vision_data: dict, name: str, category: str):
+    # Block dangerous SQL keywords
+    dangerous_keywords = ['DROP', 'DELETE', 'UPDATE', 'INSERT', 'ALTER', 'CREATE', 'TRUNCATE', 
+                        'EXEC', 'EXECUTE', 'UNION', '--', '/*', '*/', ';']
+    for keyword in dangerous_keywords:
+        if keyword in normalized_query:
+            return False, f"Query contains forbidden keyword: {keyword}"
+    
+    # Ensure query includes user_id filter for security
+    # Check if query references items table and includes user_id filter
+    if 'ITEMS' in normalized_query:
+        # Use parameterized check - username will be validated separately
+        if 'USER_ID' not in normalized_query:
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)